src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/merchandise_returns.html.twig

```diff
@@ -23,33 +23,41 @@
  * International Registered Trademark & Property of PrestaShop SA
  *#}
 
-{% if orderForViewing.returns.orderReturns is not empty %}
-  <table class="table">
-    <thead>
+{% if not orderForViewing.virtual %}
+  {% if orderForViewing.returns.orderReturns is not empty %}
+    <table class="table">
+      <thead>
       <tr>
         <th>{{ 'Date'|trans({}, 'Admin.Global') }}</th>
         <th>{{ 'Type'|trans({}, 'Admin.Global') }}</th>
         <th>{{ 'Carrier'|trans({}, 'Admin.Shipping.Feature') }}</th>
         <th>{{ 'Tracking number'|trans({}, 'Admin.Shipping.Feature') }}</th>
       </tr>
-    </thead>
-    <tbody>
-    {% for return in orderForViewing.returns.orderReturns %}
-      <tr>
-      <td>{{ return.date|date('Y-m-d H:i:s') }}</td>
-      <td>{{ return.type }}</td>
-      <td>{{ return.stateName }}</td>
-      <td>
-        {% if return.trackingNumber and return.trackingUrl %}
-          <a href="{{ return.trackingUrl }}">{{ return.trackingNumber }}</a>
-        {% elseif return.trackingNumber %}
-          {{ return.trackingNumber }}
-        {% endif %}
-      </td>
-    </tr>
-    {% endfor %}
-    </tbody>
-  </table>
+      </thead>
+      <tbody>
+      {% for return in orderForViewing.returns.orderReturns %}
+        <tr>
+          <td>{{ return.date|date('Y-m-d H:i:s') }}</td>
+          <td>{{ return.type }}</td>
+          <td>{{ return.stateName }}</td>
+          <td class="text-right">
+            {% if return.trackingNumber and return.trackingUrl %}
+              <a href="{{ return.trackingUrl }}">{{ return.trackingNumber }}</a>
+            {% elseif return.trackingNumber %}
+              {{ return.trackingNumber }}
+            {% endif %}
+          </td>
+        </tr>
+      {% endfor %}
+      </tbody>
+    </table>
+  {% else %}
+    <p class="text-center mb-0">{{ 'No merchandise returned yet'|trans({}, 'Admin.Orderscustomers.Notification') }}</p>
+  {% endif %}
+
+  {% if orderForViewing.shipping.carrierModuleInfo %}
+    {{ orderForViewing.shipping.carrierModuleInfo|raw }}
+  {% endif %}
 {% else %}
-  <p class="text-center mb-0">{{ 'No merchandise returned yet'|trans({}, 'Admin.Orderscustomers.Notification') }}</p>
+  <p class="text-center mb-0">{{ 'Merchandise returns are not applicable for virtual orders'|trans({}, 'Admin.Orderscustomers.Notification') }}</p>
```

Comment:
Is this notification displayed when an order contains only virtual products or when it contains some, among other standard products?

Reply:
This notification is only displayed on order page when an order contains only virtual products. When it contains some virtual and standard products, it is not displayed.